src/main/java/org/mockbukkit/mockbukkit/inventory/InventoryViewMock.java

```diff
@@ -145,15 +146,59 @@ public void setCursor(@Nullable ItemStack item)
 	@Override
 	public void setItem(int slot, @Nullable ItemStack item)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		int topSize = getTopInventory().getSize();
+		if (slot < 0)
+		{
+			if (item != null)
+			{
+				Location location = bottomInventory.getLocation(); // should be the player
+				if (location != null)
+				{
+					location.getWorld().dropItemNaturally(location, item);
```

Comment:
From looking at the paper code this happens:
```java
this.player.getHandle().drop(stack, false);
```
The item is thrown out of the players head, confirmed from testing. Retain ownership is probably whether it can be picked up instantly, don't know exactly. Anyhow, the items seem to also have a velocity which is relative to the players looking direction.

Reply:
From what i can see from occasional testing (remember that we have a chest open, so there shouldn't be any velocity).
I'll have a look as well, but from my tests it looks like "DropNatural"

Reply:
This code throws the item from the players head though. Also, remember to test from paper you are not, eh and yeah there might be a difference depending on inventoryview type, but from a quick search, this does not look like it's the case.
```java
@EventHandler
void onPlayerInteract(PlayerInteractEvent event) {
    event.getPlayer().getOpenInventory().setItem(- 1, new ItemStack(Material.DIAMOND));
}
```